### PR TITLE
Added Adyen template overrides

### PR DESCRIPTION
**Pull request by douglashall**
Initial implementation of an approach which would allow payment processor code to override default Otto/Oscar templates.

### CODE REVIEW

**Comment by douglashall** [comment]
FYI @zubair-arbi

File: ecommerce/settings/base.py [hunk 1]
```diff
@@ -197,6 +199,11 @@
         }
     },
 ]
+
+# Payment processor template override directories
+PAYMENT_PROCESSOR_TEMPLATE_DIRS = [
```
Comment (zubair-arbi) [inline ecommerce/settings/base.py]: @douglashall is this setting still necessary?

**Comment by zubair-arbi** [comment]
@douglashall this approach overall looks goods. I have a few questions:
1. We are moving `adyen` processor configuration in a new app `adyen`, shouldn't this be in `ecommerce/extensions/payment/processors` with other payment processors?
2. In template loaders configuration we are adding templates by getting active processors list and since for now we have custom theme for `adyen` only so its ok but should we worry about the priority  in case there will be multiple custom processor's themes?

**Comment by douglashall** [comment]
Closing in favor of https://github.com/edx/ecommerce/pull/883

**Pull request closed without merging**